feat: auto-upgrade flagd-proxy with OFO upgrades (#596)
open-feature/open-feature-operator: chart/open-feature-operator/README.md

helm upgrade --install open-feature-operator openfeature/open-feature-operator
```

> [!NOTE]

> > If you have used `flagd-proxy` provider and upgrading to OFO version `v0.5.4` or higher,
> `flagd-proxy` will be automatically upgraded to the latest supported version by the `open-feature-operator`.

Reviewer: ```suggestion
> If you upgrade to OFO `v0.5.4` or higher while using a `flagd-proxy` provider, the instance of
`flagd-proxy` will be automatically upgraded to the latest supported version by the `open-feature-operator`.
```

This upgrade will also consider your current `FeatureFlagSource` configuration and adapt

> the `flagd-proxy` Deployment accordingly.

Reviewer: for "this" you are talking about `flagd-proxy` or OFO?

helm upgrade --install open-feature-operator openfeature/open-feature-operator
```

> **Note:**

Reviewer 2: Could you please use the GitHub note syntax?

https://github.com/orgs/community/discussions/16925

If you have used `flagd-proxy` provider and upgrading to OFO version `v0.5.4` or higher,

> `flagd-proxy` will be automatically upgraded to the lastest supported version by the `open-feature-operator`.

Reviewer 2: ```suggestion
`flagd-proxy` will be automatically upgraded to the latest supported version by the `open-feature-operator`.
```
